modules/space/rendering/renderableconstellationlines.cpp

```diff
@@ -0,0 +1,449 @@
+            "Failed to open Speck file {}", std::filesystem::path(_speckFile)
+        ));
+        return false;
+    }
+
+    const float scale = static_cast<float>(toMeter(_constellationUnit));
+    double maxRadius = 0.0;
+
+    int lineIndex = 0;
+
+    // The beginning of the speck file has a header that either contains comments
+    // (signaled by a preceding '#') or information about the structure of the file
+    // (signaled by the keywords 'datavar', 'texturevar', and 'texture')
+    std::string line;
+    while (true) {
+        std::getline(file, line);
+
+        if (file.eof()) {
+            break;
+        }
+
+        // Guard against wrong line endings (copying files from Windows to Mac) causes
+        // lines to have a final \r
+        if (!line.empty() && line.back() == '\r') {
+            line = line.substr(0, line.length() - 1);
+        }
+
+        if (line.empty() || line[0] == '#') {
+            continue;
+        }
+
+        std::size_t found = line.find("mesh");
+        if (found == std::string::npos) {
+            continue;
+        }
+        else {
+            // mesh lines are structured as follows:
+            // mesh -c colorindex {
+            // colorindex is the index of the color for the mesh
+            std::stringstream str(line);
+
+            ConstellationLine constellationLine;
+            constellationLine.lineIndex = lineIndex;
+
+            std::string dummy;
+            str >> dummy; // mesh command
+            dummy.clear();
+            str >> dummy; // color index command?
+            do {
+                if (dummy == "-c") {
+                    str >> constellationLine.colorIndex; // color index command
+                }
+                dummy.clear();
+                str >> dummy;
+            } while (dummy != "{");
+
+            std::getline(file, line);
+
+            // Read the identifier
+            std::stringstream id(line);
+            std::string identifier;
+
+            id >> dummy; // id command
+            dummy.clear();
+            std::getline(id, identifier); // identifier
+            ghoul::trimWhitespace(identifier);
+            std::string name = constellationFullName(identifier);
+            if (!name.empty()) {
+                constellationLine.name = name;
+            }
+
+            // Read the number of vertices
+            std::getline(file, line);
+            std::stringstream dim(line);
+            dim >> constellationLine.numV;
+
+            // We can now read the vertices data:
+            for (int l = 0; l < constellationLine.numV; ++l) {
+                std::getline(file, line);
+                if (line.substr(0, 1) == "}") {
+                    break;
+                }
+
+                std::stringstream lineData(line);
+
+                // Try to read three values for the position
+                glm::vec3 pos;
+                bool success = true;
+                for (int i = 0; i < 3; ++i) {
```

Comment:
This for loop could be replaced with the new scn library.  See for example:  https://github.com/sgct/sgct/blob/master/src/sgct/correction/domeprojection.cpp#L51